instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go

```diff
@@ -86,7 +86,18 @@ func Middleware(service string, opts ...Option) gin.HandlerFunc {
 			opts = append(opts, oteltrace.WithAttributes(rAttr))
 		}
 		ctx, span := tracer.Start(ctx, spanName, opts...)
-		defer span.End()
```

Comment:
This behavior is already in the default SDK's `End` method:

https://github.com/open-telemetry/opentelemetry-go/blob/e8973b75b230246545cdae072a548c83877cba09/sdk/trace/span.go#L403-L420

Why is this being added here? It seems like an SDK concern not an instrumentation one.

Reply:
Looking at the [latest semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/8b4ad95d21b3d5e17fe97538f93f98e537013661/docs/http/http-spans.md#common-attributes), the `error.type` attribute should be used here to capture the panic information.

It is a known thing that our HTTP instrumentation is quite behind on semantic conventions, but I think this should be our target. Instead of having configuration to enable this, we should add the `error.type` attribute if a panic occurred always.

This will require this instrumentation to go through the process of upgrading semconv which will require a [migration path](https://github.com/open-telemetry/semantic-conventions/blob/8b4ad95d21b3d5e17fe97538f93f98e537013661/docs/http/migration-guide.md).

Reply:
sorry for the late reply - getting back to this now. i think that makes sense. i'm not familiar with these migrations, but i'll happily give it a shot. @MrAlias i may poke you for help with the migrations if i run into issues or have questions. i'll apply these changes, as well as the suggestions from @dmathieu.

Reply:
actually it looks like #5092 will take care of the migration plan. i might wait until that PR is merged and then make the changes to this package on top of that. how does that sound?